File: src/main/java/org/cobbzilla/util/system/CommandProgressMarker.java
package org.cobbzilla.util.system;

import lombok.experimental.Accessors;

import java.util.regex.Pattern;

@Accessors(chain=true)
public class CommandProgressMarker {

    private int percent;
    private Pattern pattern;
    private String line;

    @java.beans.ConstructorProperties({"percent", "pattern", "line"})
    public CommandProgressMarker(int percent, Pattern pattern, String line) {
        this.percent = percent;
        this.pattern = pattern;
        this.line = line;
    }

    public CommandProgressMarker() {
    }

    public int getPercent() {
        return this.percent;
    }

    public Pattern getPattern() {
        return this.pattern;
    }

    public String getLine() {
        return this.line;
    }

    public CommandProgressMarker setPercent(int percent) {
        this.percent = percent;
        return this;
    }

    public CommandProgressMarker setPattern(Pattern pattern) {
        this.pattern = pattern;
        return this;
    }

    public CommandProgressMarker setLine(String line) {
        this.line = line;
        return this;
    }
}

File: src/main/java/org/cobbzilla/util/system/CommandProgressMarker.kt
package org.cobbzilla.util.system

import lombok.experimental.Accessors

import java.util.regex.Pattern

@Accessors(chain = true)
class CommandProgressMarker {

    private var percent: Int = 0
    private var pattern: Pattern? = null
    private var line: String? = null

    @java.beans.ConstructorProperties("percent", "pattern", "line")
    constructor(percent: Int, pattern: Pattern, line: String) {
        this.percent = percent
        this.pattern = pattern
        this.line = line
    }

    constructor() {}

    fun getPercent(): Int {
        return this.percent
    }

    fun getPattern(): Pattern? {
        return this.pattern
    }

    fun getLine(): String? {
        return this.line
    }

    fun setPercent(percent: Int): CommandProgressMarker {
        this.percent = percent
        return this
    }

    fun setPattern(pattern: Pattern): CommandProgressMarker {
        this.pattern = pattern
        return this
    }

    fun setLine(line: String): CommandProgressMarker {
        this.line = line
        return this
    }
}
